src/structures/StageChannel.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+const GuildChannel = require('./GuildChannel');
+const Collection = require('../util/Collection');
+const Permissions = require('../util/Permissions');
+
+/**
+ * Represents a guild stage channel on Discord.
+ * @extends {GuildChannel}
+ */
+class StageChannel extends GuildChannel {
+  /**
+   * @param {*} guild The guild the stage channel is part of
+   * @param {*} data The data for the stage channel
+   */
+  constructor(guild, data) {
+    super(guild, data);
+
+    /**
+     * If the guild considers this channel NSFW
+     * @type {boolean}
+     * @readonly
+     */
+    this.nsfw = Boolean(data.nsfw);
```

Comment:
StageChannels don't have an nsfw setting.

Reply:
They have one under the hood in channel payloads even if it isn't visible in the UI.

Reply:
They do in voice channels also, but it's not present in the VoiceChannel class. Which would be inconsistent with voice-based channels.

Reply:
Good spot, will make the change now :)